### ISSUE TITLE
Memory allocation/deallocation

### ISSUE THREAD

**Issue by c0ntradicti0n** [Accenture/AmpliGraph#149]
### Description

Training some model, it allocates and deallocates big amounts of memory. Couldn't it be reusing this memory and be faster therefore?
![Selection_053](https://user-images.githubusercontent.com/46457469/66252866-dbadfc00-e760-11e9-98ff-9531f4a655d2.png)

### Steps to Reproduce

Train a bigger model with ComplEx

**Comment by sumitpai** [comment]
Which version of TensorFlow are you using? (cpu/gpu)?

**Comment by c0ntradicti0n** [comment]
I looked and found version 1.14 and I'm using cpu only.

**Comment by sumitpai** [comment]
I think you see this behaviour because tensorflow would need large amount of memory to store the gradients during backprop and it would release it after the weight updates. I think each cycle there would represent training of a batch of data.  

We will run some tests to see if it is due to some issue in the data loading pipeline or a general tensorflow behavior on CPU(during backprop). I like the suggestion of reusing the memory to speed up the process. We will consider this in our future releases and see if we can optimize the training on CPU.
